src/DotNetWorker.Grpc/NativeHostIntegration/NativeMethods.cs

```diff
@@ -8,14 +8,29 @@
 
 namespace Microsoft.Azure.Functions.Worker.Grpc.NativeHostIntegration
 {
-    internal static unsafe partial class NativeMethods
+    internal static unsafe class NativeMethods
     {
         private const string NativeWorkerDll = "FunctionsNetHost.exe";
 
+        // NET7.0 onwards, we use custom import resolver to get our native executable handle.
+        // which works for both Windows and Linux.
+#if NET7_0_OR_GREATER
+        static NativeMethods()
+        {
+            NativeLibrary.SetDllImportResolver(typeof(NativeMethods).Assembly, ImportResolver);
+        }
+#endif
+
         public static NativeHost GetNativeHostData()
         {
-            _ = get_application_properties(out var hostData);
-            return hostData;
+            var result = get_application_properties(out var hostData);
+            if (result == 1)
+            {
+                return hostData;
+            }
+
+            throw new InvalidOperationException(
+                $"Invoking get_application_properties method failed. Expected result:1, Actual result:{result}");
```

Comment:
nit: Change to `$"Invalid result returned from get_application_properties: {result}"`